### PR TITLE
Feature/10464 crypto functions part 2/10640 requesting service identity document

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Request and processing for the service identity document.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10640

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import ENASecurity
+
+protocol TVServiceIdentityDocumentProcessing {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift:8]:
```suggestion
protocol TicketValidationServiceIdentityDocumentProcessing {
```
Using TV looks like old Objective C code to me

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	func process(
+		validationServiceJwkSet: [JSONWebKey],
+		serviceIdentityDocument: TicketValidationServiceIdentityDocument,
+		completion: @escaping (Result<ServiceIdentityRequestResult, ServiceIdentityRequestError>) -> Void
+	)
+}
+
+struct TVServiceIdentityDocumentProcessor: TVServiceIdentityDocumentProcessing {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift:16]:
```suggestion
struct TicketValidationServiceIdentityDocumentProcessor: TicketValidationServiceIdentityDocumentProcessing {
```

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

**Review by pascal-brause — APPROVED** [review]

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/ServiceIdentityDocumentResource.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+struct ServiceIdentityDocumentResource: Resource {
+
+	// MARK: - Init
+
+	init(
+		endpointUrl: URL
+	) {
+		self.locator = Locator.serviceIdentityDocument(endpointUrl: endpointUrl)
```
[the remark this answers is not in the document]
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/ServiceIdentityDocumentResource.swift:23]: done

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift]: Done

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/TicketValidation/TVServiceIdentityDocumentProcessor.swift]: done

**Review by KaiTeuber — APPROVED** [review]

**Pull request merged**